src/librustc/lint/levels.rs

```diff
@@ -245,11 +246,33 @@ impl<'a> LintLevelsBuilder<'a> {
                                     }
                                 } else {
                                     let mut err = bad_attr(name_value.span);
-                                    err.help("reason must be a string literal");
+                                    if let ast::LitKind::ByteStr(_) = name_value.node {
```

Comment:
No code is good, code is evil.

Reply:
I don't think this case will ever be hit realistically, so I see this as an immediate technical debt.